File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/organisms/hero/OrgHeroPhoto.kt
package com.perrystreet.woof.designsystem.atomic.organisms.hero

import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.perrystreet.woof.designsystem.atomic.atoms.background.AtomHeroScrim
import com.perrystreet.woof.designsystem.atomic.atoms.background.AtomScrim
import com.perrystreet.woof.designsystem.atomic.molecules.image.MolAsyncImage
import com.perrystreet.woof.designsystem.atomic.molecules.image.state.AsyncImageState

@Composable
fun OrgHeroPhoto(
    imageState: AsyncImageState,
    contentDescription: String,
    dimProgress: () -> Float,
) {
    Box(modifier = Modifier.fillMaxSize()) {
        MolAsyncImage(
            state = imageState,
            contentDescription = contentDescription,
            modifier = Modifier.fillMaxSize(),
        )
        AtomHeroScrim()
        AtomScrim(alphaProvider = dimProgress)
    }
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Organisms/Hero/OrgHeroPhoto.swift
import SwiftUI

public struct OrgHeroPhoto: View {
    private let imageState: AsyncImageState
    private let contentDescription: String
    private let dimProgress: CGFloat

    public init(imageState: AsyncImageState, contentDescription: String, dimProgress: CGFloat) {
        self.imageState = imageState
        self.contentDescription = contentDescription
        self.dimProgress = dimProgress
    }

    public var body: some View {
        ZStack {
            MolAsyncImage(state: imageState, contentDescription: contentDescription)
            AtomHeroScrim()
            AtomScrim(dimProgress: dimProgress)
        }
    }
}
